Lib/multiprocessing/shared_memory.py

```diff
@@ -432,10 +546,15 @@ def __setitem__(self, position, value):
     def __len__(self):
         return struct.unpack_from("q", self.shm.buf, 0)[0]
 
+    def __repr__(self):
+        return f'{self.__class__.__name__}({list(self)}, name={self.shm.name!r})'
```

Comment:
Does `list(self)` prints all the elements in the iterable? If so, it appears it may print too much stuff on screen (and also unnecessarily make a copy in memory).

Reply:
Yes, it will copy all of the contents into a new list object.  This can be expensive and overwhelm the user in the interactive shell.  This does meet the expectations of __repr__ though.

I am familiar with some 3rd party library objects cheating and not showing everything necessary for the re-creation of the object (as repr asks us to do).  Do you think we should do something similar?  For example:  ShareableList([1, 2, 3, ...], name='blah')

Reply:
The more I think about this, the more I like the idea of removing __repr__ and offering __str__ instead which, when the length of the list is >1000 (a cutoff employed by numpy), produces output that looks like this:  ShareableList([0, 1, 2, ..., 998, 999, 1000], name='blah')

Reply:
I'm also still not so sure now. FWIW one may argue it could even `return list(self)`:

```
>>> import multiprocessing.managers
>>> mgr = multiprocessing.managers.SyncManager()
>>> mgr.start()
>>> ll = mgr.list((1,2,3))
>>> ll
[1, 2, 3]
```

The "name" parameter seems important though. I'll leave it up to you.